pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -484,7 +484,7 @@ type ComponentSettings struct {
 	Scheduler          ControllerSettings      `json:"scheduler"`
 	Etcd               EtcdStatefulSetSettings `json:"etcd"`
 	Prometheus         StatefulSetSettings     `json:"prometheus"`
-	NodePortProxyEnvoy NodeportProxyComponent  `json:"nodePortProxyEnvoy"`
+	NodePortProxyEnvoy *NodeportProxyComponent `json:"nodePortProxyEnvoy,omitempty"`
```

Comment:
just out of interest, what makes the nodeportenvoy settings different from all the others in this struct?

Reply:
Nothing really, I just didn't want touch something was there before. Shall I reflect the change on all of them ?

Reply:
If I look at dev, a sample cluster has all those fields defaulted:

```yaml
  componentsOverride:
    apiserver:
      nodePortRange: 30000-32767
      replicas: 2
    controllerManager:
      leaderElection: {}
      replicas: 1
    etcd:
      clusterSize: 3
      diskSize: 5Gi
    prometheus: {}
    scheduler:
      leaderElection: {}
      replicas: 1
```

So I wonder why the NodePort settings are not defaulted. Those settings should trickle down from the Seed to the userclusters via the code in pkg/defaulting. And the Seed gets defaulted in pkg/controller/operator/defaults.

Reply:
Ah I see, completely overlooked this part. Thanks for the hint.